placeholder-projects/buildSrc/src/main/kotlin/Build.kt

```diff
@@ -1,58 +1,68 @@
 import org.gradle.api.Named
 import org.gradle.api.Project
 import org.gradle.api.tasks.Copy
+import org.jetbrains.kotlin.gradle.plugin.KotlinDependencyHandler
 import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
 import java.io.File
 
-data class BuildStatus(val dependencies: MutableList<String> = mutableListOf())
-
 object Build {
 
+  private const val DEPENDENCY_DECLARATION = "useNpmPackage"
+
+  private const val OUTPUT_DIR_ENV = "OUTPUT_DIR"
+
+  private const val WS = "[ \\t]*"
+
   private val REACT_REGEX = noArgsFunctionCallRegex("useReact")
 
   private val COMPOSE_REGEX = noArgsFunctionCallRegex("useCompose")
 
-  private val MAIN_REGEX = Regex("fun[ \\t]main[ \\t]*[(][ \\t]*[)]")
+  private val MAIN_REGEX = Regex("fun${WS}main${WS}[(]${WS}[)]")
```

Comment:
Now it'd accept `funmain()` too. A test could be really helpful

Reply:
fixed, I will try to add tests package then to this project, this sounds like a good idea but I'll have to think how to do this properly

Reply:
The tests from buildSrc are run before every compilation of this module and if they fail the build is not continued. I didn't manage to extract them somewhere else but it seems to be a good enough approach (but there is no possibility to run them manually or I didn't find it)